CMakeLists.txt

```diff
@@ -7402,7 +7407,7 @@ IF(MSVC)
   SET_PROPERTY(SOURCE ${ALL_MICROKERNEL_SRCS} APPEND_STRING PROPERTY COMPILE_FLAGS "$<$<NOT:$<CONFIG:Debug>>: /O2 >")
   SET_PROPERTY(SOURCE ${HOT_SRCS} APPEND_STRING PROPERTY COMPILE_FLAGS "$<$<NOT:$<CONFIG:Debug>>: /O2 >")
   SET_PROPERTY(SOURCE ${COLD_SRCS} APPEND_STRING PROPERTY COMPILE_FLAGS "$<$<NOT:$<CONFIG:Debug>>: /O1 >")
-ELSE()
+ELSEIF(NOT XCODE)
```

Comment:
With this change, optimization settings are not set at all for the compiled files, resulting in subpar performance. A proper solution would be to refactor CMakeLists so that the per-config flags can be set at a target level.

Reply:
Done.

Using CMAKE_BUILD_TYPE to check its buid-type and decide if we should add those optimization flags.